README.md

```diff
@@ -1,7 +1,13 @@
 # Solo.io APIs
 `gloo-repo-branch` serves to provide the APIs required for Gloo OSS, such as Rate Limiter APIs. Instead of using Gloo LTS
-releases to update APIs, this branch is used by Gloo to get updated API information. Therefore we do not need to sync the APIs. 
+releases to update APIs, this branch is used by Gloo to get updated API information. Therefore we do not need to sync the APIs.
 It is not intended to be updated as much as the regular LTS Solo APIs.
 
 This branch stemmed off of the [gloo-edge-safe-hasher](https://github.com/solo-io/solo-apis/tree/gloo-edge-safe-hasher) tag which is used
-in Gloo OSS due to its minimized hashing which prevents data races in gloo. [Original Issue](https://github.com/solo-io/gloo/issues/7209).
\ No newline at end of file
+in Gloo OSS due to its minimized hashing which prevents data races in gloo. [Original Issue](https://github.com/solo-io/gloo/issues/7209).
+
+Any changes to APIs in [rate-limiter](https://github.com/solo-io/rate-limiter/) should propagate to this branch and code-gen should be re-run. There should be [existing automation](https://github.com/solo-io/rate-limiter/actions/workflows/update-solo-apis.yaml) around this but it is always a good practice to manually sync the protos when bumping k8s dependencies or cutting a major release in case the automation was unsuccessful. Any CRDs generated / modified can be ignored since we only care about the protos.
+
+An alternative that had been considered is using `gloo-main` instead of this branch in Gloo OSS. However this poses an issue when building enterprise - if `gloo-main` is ahead of the latest `gloo-v1.n.x tag` used in enterprise, enterprise will utilize solo-apis@gloo-main instead of the specified tag due to the way go bundles dependencies.
```

Comment:
Can you help me understand this a bit better? We have a requirement in solo-projects, where the gloo dependency and solo-apis dependency must be in lockstep: https://github.com/solo-io/solo-projects/blob/main/Makefile#L278

The gloo-main branch will always hold the latest tags of gloo edge, the betas of the current release. 

Perhaps I'm misunderstanding the limitation though. Could you include an example so I can better understand?

Reply:
There can be situations where commits have been pushed to `gloo-main` or PRs have been created that do not include tags ([example 1](https://github.com/solo-io/solo-apis/pull/1082), [example 2](https://github.com/solo-io/solo-apis/pull/1081)) which can cause gloo-main to be ahead of the latest tag - albeit quite rare.
If gloo OSS depends on gloo-main which is ahead of the latest tag, when solo-projects imports solo-apis despite specifying the tag, the latest version of the dependency is used (in this case gloo-main). I'm unable to find such a scenario mentioned in the docs but have faced this issue when trying it locally.

Reply:
`If gloo OSS depends on gloo-main which is ahead of the latest tag` - Does this ever happen? Shouldn't Gloo always depend on a tagged version?

Reply:
i assume this is referring to the below scenario where gloo repo depends on the `gloo-main` branch (instead of the `gloo-repo-branch` as it does now).

we could do this in 2 ways: 
1. gloo depends on `gloo-main` (latest commit). gloo-main can get ahead of the latest gloo-v1.n.x tag if we've merged manual changes into gloo-main (see david's examples above) and haven't yet done a gloo release which creates a tag that includes those changes. this does happen sometimes. it may be undesirable if solo-projects is not yet ready to consume those changes for whatever reason
2. gloo depends on a specific solo-apis tag (gloo-v1.n.x). this won't automatically pick up ratelimiter api changes that are pushed to gloo-main. the steps for gloo to pick up those changes would be:
    1. do a ratelimiter release
    2. let the workflow run that pushes the ratelimiter api changes to solo-apis (or do that manually if workflow is still broken)
    3. release gloo so a solo-apis tag can be created, that contains the ratelimiter api updates (this may not be desirable if we are not actually ready to release gloo, or there are no pending gloo changes that need to be released)
    4. update gloo's go.mod to pick up the solo-apis tag (gloo-v1.n.x)

Reply:
Awesome! I think this explanation, flushing out the particulars as you do will help us retain this information. Can we update the readme to incorporate some of these details?